src/pages/ConnectWallet/components/WalletList.tsx

```diff
@@ -0,0 +1,168 @@
+      const deviceId = item?.id
+      if (adapter && deviceId) {
+        const { name, icon } = MobileConfig
+        try {
+          const revoker = await getWallet(deviceId)
+          if (!revoker?.mnemonic) throw new Error(`Mobile wallet not found: ${deviceId}`)
+          if (!revoker?.id) throw new Error(`Revoker ID not found: ${deviceId}`)
+
+          const wallet = await adapter.pairDevice(revoker.id)
+          await wallet?.loadDevice({ mnemonic: revoker.mnemonic })
+          if (!(await wallet?.isInitialized())) {
+            await wallet?.initialize()
+          }
+          dispatch({
+            type: WalletActions.SET_WALLET,
+            payload: {
+              wallet,
+              name,
+              icon,
+              deviceId,
+              meta: { label: item.label },
+              connectedType: KeyManager.Mobile,
+            },
+          })
+          dispatch({ type: WalletActions.SET_IS_CONNECTED, payload: true })
+          dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: false })
+          dispatch({
+            type: WalletActions.SET_CONNECTOR_TYPE,
+            payload: { modalType: KeyManager.Mobile, isMipdProvider: false },
+          })
+
+          localWallet.setLocalWallet({ type: KeyManager.Mobile, deviceId })
+          localWallet.setLocalNativeWalletName(item?.label ?? 'label')
+        } catch (e) {
+          console.log(e)
+          setError('walletProvider.shapeShift.load.error.pair')
+        }
+      } else {
+        setError('walletProvider.shapeShift.load.error.pair')
+      }
+    },
+    [dispatch, getAdapter, localWallet],
+  )
+
+  const handleRename = useCallback(
+    (wallet: RevocableWallet) => {
+      history.push(MobileWalletDialogRoutes.RENAME, { vault: wallet })
+    },
+    [history],
+  )
+
+  const handleDelete = useCallback(
+    (wallet: RevocableWallet) => {
+      history.push(MobileWalletDialogRoutes.DELETE, { vault: wallet })
+    },
+    [history],
+  )
+
+  const content = useMemo(() => {
+    if (error) {
+      return (
+        <Alert status='error' borderRadius='md'>
+          <AlertIcon />
+          <AlertDescription>{error}</AlertDescription>
+        </Alert>
+      )
+    }
+    return (
+      <Stack>
+        {wallets?.map(wallet => {
+          const isSelected = walletInfo?.deviceId === wallet.id
+          const _hover = isSelected
+            ? { bg: 'background.button.secondary.base', opacity: '1' }
+            : undefined
+          const _active = isSelected
+            ? { bg: 'background.button.secondary.base', opacity: '1' }
+            : undefined
+          return (
+            <WalletCard
+              id={wallet.id}
+              key={wallet.id}
+              wallet={wallet}
+              onClick={isSelected ? undefined : handleWalletSelect}
+              isActive={isSelected}
+              isEditing={isEditing}
+              onRename={handleRename}
+              onDelete={handleDelete}
```

Comment:
Disable if `isSelected`.

Reply:
I didn't disable so we have the same style as usually but prevented all the callbacks to be called is isSelected

In [2d9f44b](https://github.com/shapeshift/web/pull/8689/commits/2d9f44b23a0ce70c448c2fbcfaf53654c99ef88f) and [3d4afb4](https://github.com/shapeshift/web/pull/8689/commits/3d4afb44eed27599a5e9015ec6cf45a6ca4ce0a3)